pkg/kubectl/resource_printer.go

```diff
@@ -1198,3 +1199,85 @@ func indirect(v reflect.Value) (rv reflect.Value, isNil bool) {
 	}
 	return v, false
 }
+
+// Sorting printer sorts list types before delegating to another printer.
+// Non-list types are simply passed through
+type SortingPrinter struct {
+	SortField string
+	Delegate  ResourcePrinter
+}
+
+type RuntimeSort struct {
+	field string
+	objs  []runtime.Object
+}
+
+func (r *RuntimeSort) Len() int {
+	return len(r.objs)
+}
+
+func (r *RuntimeSort) Swap(i, j int) {
+	r.objs[i], r.objs[j] = r.objs[j], r.objs[i]
+}
+
+func (r *RuntimeSort) Less(i, j int) bool {
+	iObj := r.objs[i]
+	jObj := r.objs[j]
+
+	parser := jsonpath.New("sorting")
+	parser.Parse(r.field)
+
+	iValues, err := parser.FindResults(reflect.ValueOf(iObj).Elem().Interface())
+	if err != nil {
+		glog.Fatalf("Failed to get i values for %#v using %s (%#v)", iObj, r.field, err)
```

Comment:
Can we put a check that the path exists for the object(s) at the beginning of `sortObj`, and surface an error if it doesn't? This is not a friendly error to get if a user misplaces a period or a capital in a fieldpath.